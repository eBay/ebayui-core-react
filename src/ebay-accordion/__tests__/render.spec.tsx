import React from 'react'
import { composeStories } from '@storybook/react'

import * as stories from './index.stories'
import { render } from '@testing-library/react'

const { Default, Large, Open, AutoCollapse } = composeStories(stories)

describe('<EbayAccordion /> rendering', () => {
    it('renders default story correctly', () => {
        const { container } = render(<Default />)
        expect(container).toMatchSnapshot()
    })

    it('renders large size story correctly', () => {
        const { container } = render(<Large />)
        expect(container).toMatchSnapshot()
    })

    it('renders open story correctly', () => {
        const { container } = render(<Open />)
        expect(container).toMatchSnapshot()
    })

    it('renders autoCollapse story correctly', () => {
        const { container } = render(<AutoCollapse />)
        expect(container).toMatchSnapshot()
    })
})
