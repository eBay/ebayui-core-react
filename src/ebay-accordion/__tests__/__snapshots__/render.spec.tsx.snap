// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayAccordion /> rendering renders autoCollapse story correctly 1`] = `
<div>
  <ul
    aria-roledescription="accordion"
    class="accordion"
  >
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 1
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 2
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 3
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>
`;

exports[`<EbayAccordion /> rendering renders default story correctly 1`] = `
<div>
  <ul
    aria-roledescription="accordion"
    class="accordion"
  >
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 1
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 2
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 3
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>
`;

exports[`<EbayAccordion /> rendering renders large size story correctly 1`] = `
<div>
  <ul
    aria-roledescription="accordion"
    class="accordion accordion--large"
  >
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 1
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 2
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 3
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>
`;

exports[`<EbayAccordion /> rendering renders open story correctly 1`] = `
<div>
  <ul
    aria-roledescription="accordion"
    class="accordion"
  >
    <li>
      <details
        class="details"
        open=""
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 1
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 2
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
    <li>
      <details
        class="details"
      >
        <summary
          class="details__summary"
        >
          <span
            class="details__label"
          >
            Item 3
          </span>
          <span
            class="details__icon"
            hidden=""
          >
            <svg
              aria-hidden="true"
              class="details__expand icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-add-16"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="details__collapse icon icon--16"
              focusable="false"
              xmlns="http://www.w3.org/2000/svg"
            >
              <use
                xlink:href="#icon-remove-16"
              />
            </svg>
          </span>
        </summary>
        <div
          class="details__content"
        >
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
      </details>
    </li>
  </ul>
</div>
`;
